syntax = "proto3";

package tea.core;
option java_package = 'com.neuromancer42.tea.core';

service CoreService {
    rpc RunAnalyses (ApplicationRequest) returns (stream ApplicationResponse) {}
}

message Compilation {
    string source = 1;
    string command = 2;
}

message Test {
    repeated string arg = 1;
    optional string test_id = 2;
}

message ApplicationRequest {
    string project_id = 1;
    map<string, string> option = 2;
    Compilation source = 3;
    repeated string analysis = 4;
    repeated string alarm_rel = 5;
    optional bool need_rank = 6;
    repeated Test test_suite = 7;
}

message ApplicationResponse {
    optional string msg = 2;
    repeated string alarm = 3;
}